cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -731,6 +733,11 @@ getPkgConfigDb verbosity progdb = do
     liftIO $ readPkgConfigDb verbosity progdb
 
 
+getDirectoryContentsMonitored :: FilePath -> Rebuild [FilePath]
+getDirectoryContentsMonitored dir = do
+    monitorFiles [monitorDirectory dir]
```

Comment:
This behaviour is the same on NTFS, though apparently not on FAT.

https://support.microsoft.com/en-us/kb/299648